script/utils/update-redirects-data.ts

```diff
@@ -60,10 +69,20 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
       countNewRedirections++;
     // remove `else` to enable legitimately removing pages from books
     // only once uuids are guaranteed to be consistent
-    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException) {
+    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException && !allowBookRedirect) {
       throw new Error(
         `updateRedirects prohibits removing pages from a book, `
         + `but neither section with ID ${section.id} nor slug ${section.slug} was found in book ${newBook.id}`);
+    } else if (allowBookRedirect && archiveTreeSectionIsPage(section)) {
```

Comment:
@TomWoodward I probably need more conditions here such as checking for existing redirects like on line 51/60. I'm also wondering if there's a better way to exclude chapter and unit slugs than `archiveTreeSectionIsPage`.

Reply:
I....I feel like we aren't expecting the array to have non page entries in it, so maybe that would be a better change... Maybe